test/unit/EncryptionUtil.test.js

```diff
@@ -0,0 +1,171 @@
+            ['streamId', 0, 1, 0, 'publisherId', 'msgChainId'], null,
+            StreamMessage.CONTENT_TYPES.MESSAGE, StreamMessage.ENCRYPTION_TYPES.NONE, {
+                foo: 'bar'
+            }, StreamMessage.SIGNATURE_TYPES.NONE, null,
+        )
+        EncryptionUtil.encryptStreamMessage(streamMessage, key)
+        const newKey = EncryptionUtil.decryptStreamMessage(streamMessage, key)
+        assert.deepStrictEqual(newKey, null)
+        assert.deepStrictEqual(streamMessage.getSerializedContent(), '{"foo":"bar"}')
+        assert.deepStrictEqual(streamMessage.encryptionType, StreamMessage.ENCRYPTION_TYPES.NONE)
+    })
+    it('StreamMessage gets encrypted with new key', () => {
+        const key = crypto.randomBytes(32)
+        const newKey = crypto.randomBytes(32)
+        const streamMessage = StreamMessage.create(
+            ['streamId', 0, 1, 0, 'publisherId', 'msgChainId'], null,
+            StreamMessage.CONTENT_TYPES.MESSAGE, StreamMessage.ENCRYPTION_TYPES.NONE, {
+                foo: 'bar'
+            }, StreamMessage.SIGNATURE_TYPES.NONE, null,
+        )
+        EncryptionUtil.encryptStreamMessageAndNewKey(newKey, streamMessage, key)
+        assert.notDeepStrictEqual(streamMessage.getSerializedContent(), '{"foo":"bar"}')
+        assert.deepStrictEqual(streamMessage.encryptionType, StreamMessage.ENCRYPTION_TYPES.NEW_KEY_AND_AES)
+    })
+    it('StreamMessage decryption after encryption equals the initial StreamMessage (with new key)', () => {
+        const key = crypto.randomBytes(32)
+        const newKey = crypto.randomBytes(32)
+        const streamMessage = StreamMessage.create(
+            ['streamId', 0, 1, 0, 'publisherId', 'msgChainId'], null,
+            StreamMessage.CONTENT_TYPES.MESSAGE, StreamMessage.ENCRYPTION_TYPES.NONE, {
+                foo: 'bar'
+            }, StreamMessage.SIGNATURE_TYPES.NONE, null,
+        )
+        EncryptionUtil.encryptStreamMessageAndNewKey(newKey, streamMessage, key)
+        const newKeyReceived = EncryptionUtil.decryptStreamMessage(streamMessage, key)
+        assert.deepStrictEqual(newKeyReceived, newKey)
+        assert.deepStrictEqual(streamMessage.getSerializedContent(), '{"foo":"bar"}')
+        assert.deepStrictEqual(streamMessage.encryptionType, StreamMessage.ENCRYPTION_TYPES.NONE)
+    })
+    it('throws if invalid public key passed in the constructor', () => {
+        const keys = crypto.generateKeyPairSync('rsa', {
+            modulusLength: 4096,
+            publicKeyEncoding: {
+                type: 'pkcs1',
+                format: 'pem',
+            },
+            privateKeyEncoding: {
+                type: 'pkcs1',
+                format: 'pem',
+            },
+        })
+        assert.throws(() => {
+            // eslint-disable-next-line no-new
+            new EncryptionUtil({
+                privateKey: keys.privateKey,
+                publicKey: 'wrong public key',
+            })
+        }, /Error/)
```

Comment:
This `/Error/` validation.doesn't do much since almost every Error will match `/Error/`. Perhaps should update it to match against the actual expected error, or do nothing instead.